src/utils.ts

```diff
@@ -38,14 +38,16 @@ export function findSMDataSources(): Array<DataSourceInstanceSettings<SMOptions>
   }) as unknown as Array<DataSourceInstanceSettings<SMOptions>>;
 }
 
-export function findLinkedDatasource(linkedDSInfo: LinkedDatasourceInfo): DataSourceInstanceSettings {
-  if (linkedDSInfo.uid) {
-    const linkedDS = Object.values(config.datasources).find((ds) => ds.uid === linkedDSInfo.uid);
+export function findLinkedDatasource(uid: string, grafanaName: string): DataSourceInstanceSettings | undefined {
+  if (uid) {
```

Comment:
I don't think we ever have a situation where when we use this function the uid _isn't_ present. The next PR I am going to be reworking the app initialiser process so will make a decision if this is needed.